tests/coverage/coverage.py

```diff
@@ -0,0 +1,101 @@
+# SPDX-License-Identifier: Apache-2.0
```

Comment:
let's output coverage as structured data. That way we might be able to build reports in the future and that is disjunct from calculating the coverage (e.g. we might do with off-the-shelf tools like jq)